src/app/features/public-form/public-form.component.spec.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  ComponentFixture,
+  fakeAsync,
+  TestBed,
+  tick,
+} from "@angular/core/testing";
+
+import { PublicFormComponent } from "./public-form.component";
+import { MockedTestingModule } from "../../utils/mocked-testing.module";
+import { PouchDatabase } from "../../core/database/pouch-database";
+import { PublicFormConfig } from "./public-form-config";
+import { LoginState } from "../../core/session/session-states/login-state.enum";
+import { ActivatedRoute } from "@angular/router";
+import { Child } from "../../child-dev-project/children/model/child";
+import { genders } from "../../child-dev-project/children/model/genders";
+import { MatSnackBar } from "@angular/material/snack-bar";
+import { EntityFormService } from "../../core/entity-components/entity-form/entity-form.service";
+import { ConfigService } from "../../core/config/config.service";
+
+describe("PublicFormComponent", () => {
+  let component: PublicFormComponent<Child>;
+  let fixture: ComponentFixture<PublicFormComponent<Child>>;
+  let initIndexedDBSpy: jasmine.Spy;
+  const formConfigID = "form-id";
+  let initConfig: () => void;
+
+  beforeEach(async () => {
+    const formConfig = new PublicFormConfig(formConfigID);
+    formConfig.title = "test form";
+    formConfig.entity = "Child";
+    formConfig.columns = [["name"], ["gender"]];
+    formConfig.prefilled = { status: "new", gender: "M" };
+    await TestBed.configureTestingModule({
+      imports: [
+        PublicFormComponent,
+        MockedTestingModule.withState(LoginState.LOGGED_IN, [formConfig]),
+      ],
+      providers: [
+        {
+          provide: ActivatedRoute,
+          useValue: { snapshot: { paramMap: new Map([["id", formConfigID]]) } },
+        },
+      ],
+    }).compileComponents();
+
+    initIndexedDBSpy = spyOn(TestBed.inject(PouchDatabase), "initIndexedDB");
+    initConfig = () => {
+      const configService = TestBed.inject(ConfigService);
+      configService["_configUpdates"].next(configService["currentConfig"]);
+    };
+
+    fixture = TestBed.createComponent(PublicFormComponent<Child>);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it("should create", () => {
+    expect(component).toBeTruthy();
+  });
+
+  it("should initialize remote DB on startup", () => {
+    expect(initIndexedDBSpy).toHaveBeenCalled();
+  });
+
+  it("should initialize component with values from PublicFormConfig once config is ready", fakeAsync(() => {
+    expect(component.entity).toBeUndefined();
+
+    initConfig();
+    tick();
+
+    expect(component.entity.getConstructor()).toBe(Child);
```

Comment:
would be a bit easier to follow if you set the relevant formConfig property in this test here